c2rust-analyze/src/c_void_casts.rs

```diff
@@ -158,25 +162,33 @@ pub struct CVoidCast<'tcx> {
 /// to their properly typed pointers,
 /// like [`CVoidCasts`], but in a single direction, meaning it represents either
 /// [`CVoidCastDirection::From`] or [`CVoidCastDirection::To`].
-#[derive(Default)]
-pub struct CVoidCastsUniDirectional<'tcx>(HashMap<CVoidPtr<'tcx>, Place<'tcx>>);
+#[derive(Default, Clone, Debug)]
+pub struct CVoidCastsUniDirectional<'tcx>(HashMap<Location, (CVoidPtr<'tcx>, Place<'tcx>)>);
 
 impl<'tcx> CVoidCastsUniDirectional<'tcx> {
-    pub fn contains(&self, place: Place<'tcx>) -> bool {
-        self.0.contains_key(&place)
+    pub fn contains(&self, loc: &Location) -> bool {
+        self.0.contains_key(loc)
     }
```

Comment:
For here and other places, you can just take `Location` by value as it's `Copy`.

Reply:
7fb2d448